celery_batches/__init__.py

```diff
@@ -289,6 +288,21 @@ def task_message_handler(message, body, ack, reject, callbacks, **kw):
 
         return task_message_handler
 
+    def apply(self, args=None, kwargs=None, *_args, **_kwargs):
+        if self._get_app().conf.task_always_eager:
```

Comment:
Quick question about this, looking at the corresponding code in Celery's `Task` class (see https://github.com/celery/celery/blob/d0563058f8f47f347ac1b56c44f833f569764482/celery/app/task.py#L729-L780), it seems that `apply` is meant to always return an `EagerResult`. This if-statement does, however, exist as part of `apply_async` method: https://github.com/celery/celery/blob/d0563058f8f47f347ac1b56c44f833f569764482/celery/app/task.py#L545-L562

Is this if-statement necessary here? It seems like we should just return `SimpleRequest`? I might be missing something in the way the strategy calls back into `apply()` though...

Reply:
This `if`-statement is potentially tautologous, yes... I haven't been able to satisfy myself that it isn't called elsewhere, without the `task_always_eager` guard.

On the other hand, having been away and giving it a fresh look now: Even if it were called elsewhere, surely we still want to ensure that the original args/kwargs are wrapped as a `SimpleRequest` before calling the actual task... So yeah, lets get rid of the `if`!

Reply:
Cool! 👍 I did some testing with our application and this seems to work fine! Sorry for the delay.

Reply:
Got rid of the `if` 😄